### PR TITLE
remove get_form_ids_or_querystring to avoid double check on request a…

### PR DESCRIPTION
**Pull request by mkangia**
…nd then on return type

Follow up from https://github.com/dimagi/commcare-hq/pull/14206/files#diff-9f0c2d0ac3b32ff875ff48f73f7b3e10R532
@esoergel

### CODE REVIEW

File: corehq/apps/data_interfaces/views.py [hunk 1]
```diff
@@ -531,10 +521,11 @@ def inaccessible_forms_accessed(self, xform_ids, domain, couch_user):
         return xforms_user_ids - accessible_user_ids
 
     def get_xform_ids(self, request):
-        form_ids_or_query_string = self.get_form_ids_or_query_string(request)
-        if type(form_ids_or_query_string) == list:
-            xform_ids = form_ids_or_query_string
-        elif isinstance(form_ids_or_query_string, basestring):
+        if 'select_all' in self.request.POST:
+            # Altough evaluating form_ids and sending to task would be cleaner,
+            # heavier calls should be in an async task instead
```
Comment (esoergel) [inline corehq/apps/data_interfaces/views.py]: hmm, I just took a closer look at what this existing code (and this comment) were doing - the comment refers to behavior that was just changed in the last PR (the "heavier call" no longer happens in the async task).

Also, it's out of scope to change, but that request mocking business seems like a needlessly complex way of performing an elasticsearch query.

**Comment by esoergel** [comment]
Thanks for the change!

File: corehq/apps/data_interfaces/views.py (hunk 1 above)
Comment (mkangia) [inline corehq/apps/data_interfaces/views.py]: Thanks @esoergel. Happy to make any improvements in this. What do you suggest?

Comment (esoergel) [inline corehq/apps/data_interfaces/views.py]: I was just thinking of dropping the comment since it's no longer accurate.

Comment (esoergel) [inline corehq/apps/data_interfaces/views.py]: The bigger fix would be to abstract out the ES query that accepts a querystring and returns form ids, then just call it from here and from the view where it lives now.  That's a bit out of scope, and there may be complications which led the original author of this to choose this request mocking business instead, so I wouldn't worry about it too much.

**Pull request merged**